crates/vm/levm/src/precompiles.rs

```diff
@@ -173,11 +174,27 @@ fn identity(
 }
 
 fn sha2_256(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+    calldata: &Bytes,
+    gas_for_call: U256,
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    let data_size: u64 = calldata
+        .len()
+        .try_into()
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let cost = sha2_256_cost(data_size)?;
+    if gas_for_call < cost {
+        return Err(VMError::PrecompileError(PrecompileError::NotEnoughGas));
+    }
+
+    *consumed_gas = consumed_gas
+        .checked_add(cost)
+        .ok_or(PrecompileError::GasConsumedOverflow)?;
```

Comment:
Shouldn't this be an internal error in this scenario? It shouldn't ever happen, right?

Reply:
In my opinion it can happen because we pass a pointer to the consumed gas, although it is as possible as the other uses of GasConsumedOverflow throughout the vm (that should not happen since they should be lower than gas limit)